pandas/tests/dtypes/test_dtypes.py

```diff
@@ -263,10 +265,12 @@ def test_hash_vs_equality(self, dtype):
         assert dtype2 != dtype4
         assert hash(dtype2) != hash(dtype4)
 
-    def test_construction(self):
-        msg = "DatetimeTZDtype only supports ns units"
-        with pytest.raises(ValueError, match=msg):
-            DatetimeTZDtype("ms", "US/Eastern")
+    def test_construction_non_nanosecond(self):
+        res = DatetimeTZDtype("ms", "US/Eastern")
```

Comment:
Is the `DatetimeTZDtype("D", "US/Eastern")` raising case tested in this file as well?

Also would be good to have a full smoke test from one of the 1D constructors e.g. `pd.array('2022-01-01', dtype='datetime64[us, US/Eastern]')`